src/persist-client/src/internal/compact.rs

```diff
@@ -86,18 +87,30 @@ pub struct Compactor<K, V, T, D> {
     _phantom: PhantomData<fn() -> D>,
 }
 
+impl<K, V, T, D> Clone for Compactor<K, V, T, D> {
```

Comment:
Should be equivalent to the derived impl, but avoids the overly-tight default bounds.